### PR TITLE
Added Custom Amount Notifications (Fixed)

### PR DESCRIPTION
**Pull request by jwhamilton99**
Integrates all changes from pull request #42. Closes issue #40.

Integrated properly with David's changes to the notification system. Also improved some significant amount notification logic.

### CODE REVIEW

File: Shared/Controller/NotificationSettingsController.swift [hunk 1]
```diff
@@ -53,6 +62,57 @@ class NotificationSettingsController: ObservableObject {
...
+                self.rejectedInputShowing = false
+            } else {
+                //currency formatter (e.g. $300,000)
+                let f = NumberFormatter()
+                f.numberStyle = .currency
+                if let number = f.number(from: newValue) {
+                    DispatchQueue.main.async {
+                        self.rejectedInputShowing = false
+                        UserDefaults.shared.set(number.doubleValue, forKey: "customNotificationAmount")
+                        print(UserDefaults.shared.double(forKey: "customNotificationAmount"))
```
Comment (dwrss) [inline Shared/Controller/NotificationSettingsController.swift:80]: I'd have a look in Loggers.swift and user an existing logger or make a new one?

File: Shared/Controller/NotificationSettingsController.swift [hunk 2]
```diff
@@ -53,6 +62,57 @@ class NotificationSettingsController: ObservableObject {
...
+                            }
+                        } else {
+                            UserDefaults.shared.set(nil, forKey: "customNotificationAmount")
+                        }
+                    }
+                    break
+                default:
+                    UNUserNotificationCenter.current().requestAuthorization(options: [.alert, .sound], completionHandler: {(authorized, error) in
+                        if let e = error {
+                            print(e.localizedDescription)
```
Comment (dwrss) [inline Shared/Controller/NotificationSettingsController.swift:109]: Same as above RE loggers

File: Shared/Controller/NotificationSettingsController.swift [hunk 3]
```diff
@@ -53,6 +62,57 @@ class NotificationSettingsController: ObservableObject {
...
+                    }
+                } else {
+                    self.rejectedInputShowing = true
+                }
+            }
+        }).store(in: &self.subscribers)
+        
+        
+        self.$enableCustomAmountNotification.sink(receiveValue: {newValue in
+            UNUserNotificationCenter.current().getNotificationSettings(completionHandler: {(settings) in
```
Comment (dwrss) [inline Shared/Controller/NotificationSettingsController.swift:90]: This looks pretty close to what `setNotificationPreference` is doing?

File: Shared/Helpers/CheckSignificantAmounts.swift [hunk 4]
```diff
@@ -9,21 +9,31 @@ import Foundation
 
 func checkSignificantAmounts(for widgetData: TiltifyWidgetData) {
     //check if cached amount is significant
-    let cachedTotal = round((UserDefaults.shared.object(forKey: "cachedTotalRaised") as? Double) ?? -1.0)
-    let totalRaisedRaw = widgetData.totalRaised ?? -1.0
-    let totalRaised = round(totalRaisedRaw)
+    //uncomment below for testing
+//    UserDefaults.shared.set(49990.0, forKey: "cachedTotalRaised")
+    let cachedTotal = (UserDefaults.shared.object(forKey: "cachedTotalRaised") as? Double) ?? -1.0
```
Comment (dwrss) [inline Shared/Helpers/CheckSignificantAmounts.swift:14]: `UserDefaults.shared.double` might be better?

File: Shared/Helpers/CheckSignificantAmounts.swift [hunk 5]
```diff
@@ -9,21 +9,31 @@ import Foundation
...
+    
+    
+    let totalRaised = widgetData.totalRaised ?? -1.0
+    //uncomment below for testing
+//    let totalRaised = 50001.0
     let goal = widgetData.goal ?? -1.0
     
+    let customAmount = (UserDefaults.shared.object(forKey: "customNotificationAmount") as? Double) ?? -1.0
```
Comment (dwrss) [inline Shared/Helpers/CheckSignificantAmounts.swift:22]: `UserDefaults.shared.double` might be better?

**Review by dwrss — CHANGES_REQUESTED** [review]

File: Shared/Controller/NotificationSettingsController.swift (hunk 1 above)
Comment (jwhamilton99) [reply Shared/Controller/NotificationSettingsController.swift:80]: i added this for debugging, i forgot to remove it

File: Shared/Helpers/CheckSignificantAmounts.swift (hunk 4 above)
Comment (jwhamilton99) [reply Shared/Helpers/CheckSignificantAmounts.swift:14]: i forgot to switch this as well

**Review by dwrss — APPROVED** [review]

**Pull request merged**